### PR TITLE
[MOM][Aftershock] Adds new 'uses_energy' search data and converts electron overflow powers to use it

### PR DESCRIPTION
**Pull request by b3brodie**
…low powers to use that for searching.  Deletes old flag that used to be used for this.

#### Summary
Mods "Expand Electron Overflow valid targets"

#### Purpose of change
Currently, Electron Overflow (and the electric sucking monster) operate off of a couple of flags.  Unfortunately, many devices that should be chargeable with this power do not have easily useable flags, necessitating a new special "ELECTROKINETIC_CHARGEABLE" flag to mark these cases.  Unfortunately, this flag has to be manually applied to every item that doesn't have some other flags.  This is a lot of manual work that typically misses items, and also doesn't work for other mods unless specifically compatible.  Switching over to just checking if the item uses energy is much easier.

#### Describe the solution
Implement new "uses_energy" boolean for search_data.  This can takes either true or false, and will return items that either do or do not use energy.
Use "uses_energy":true for the electrokinetic powers and remove the old flags.
Delete all the old references to the ELECTROKINETIC_CHARGEABLE flag

#### Describe alternatives you've considered
Implementing a ammo type search term and searching for energy ammo type.  Unfortunately, looking through item.cpp there wasn't any good way of doing this I could see.  Methods like ammo_type() are specialized for specific item types and circumstances.  I gave it a look but wasn't able to find a good way to universally check for ammo type.  Decided to implement it this way instead since it still serves as required.

#### Testing
Created a world with mom, aftershock, and xedra.
Observed no loading errors from flag changes
Observed that electron overflow powers from both mods still charge items
Checked that some of the old items that didn't natively work are still charged.

#### Additional context
N/A

### CODE REVIEW

File: src/npctalk.cpp [hunk 1]
```diff
@@ -185,6 +185,7 @@ struct item_search_data {
     std::vector<str_or_var> material;
     std::vector<str_or_var> flags;
     std::vector<str_or_var> excluded_flags;
+    std::vector<bool> uses_energy;
```
Comment (GuardianDll) [inline src/npctalk.cpp:188]: there is no need to make it vector, just make it a bool

Comment (GuardianDll) [reply src/npctalk.cpp:188]: then the documentation need to be updated to reflect it - it was not very obvious for me

File: doc/EFFECT_ON_CONDITION.md [hunk 2]
```diff
@@ -4436,6 +4436,7 @@ search_data is an array, that allow to filter specific items from the list. At t
 | "flags" | string, [variable object](#variable-object) or array of strings or variable objects | filter the list of items by flags they have |
 | "excluded_flags" | string, [variable object](#variable-object) or array of strings or variable objects | excludes items from the list by flags they have |
 | "material" | string, [variable object](#variable-object) or array of strings or variable objects | filter the list of items by their material |
+| "uses_energy" | boolean | filter the list of items by whether or not they use energy |
```
Comment (GuardianDll) [inline doc/EFFECT_ON_CONDITION.md:4439]:
```suggestion
| "uses_energy" | boolean | filter the list of items by whether or not they use energy. `true` would pick only items that use energy, `false` would pick all items that do not use energy |
```

File: src/npctalk.cpp (hunk 1 above)
Comment (b3brodie) [reply src/npctalk.cpp]: swapped to optional

**Review by GuardianDll — APPROVED** [review]

**Pull request merged**